### PR TITLE
RDR node failure tests

### PR DESCRIPTION
**Pull request by sidhant-agrawal**
Test to verify failure of node hosting pods of different components in Regional-DR solution

Signed-off-by: Sidhant Agrawal <sagrawal@redhat.com>

### CODE REVIEW

File: tests/disaster-recovery/regional-dr/test_node_failure.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            wait_for_nodes_status([resource_node.name])
+            logger.info(
+                "Wait for all the pods in openshift-storage to be in running state"
+            )
+            assert wait_for_pods_to_be_running(
+                timeout=600
+            ), "Not all the pods reached running state."
+            dr_helpers.wait_for_mirroring_status_ok(timeout=600)
+
+        logger.info("Inducing node failure in both managed clusters simultaneously")
```
Comment (keesturam) [inline tests/disaster-recovery/regional-dr/test_node_failure.py:122]: Pls consider adding this scenario as a separate test case

File: tests/disaster-recovery/regional-dr/test_node_failure.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                    condition=constants.STATUS_RUNNING,
+                    selector=self.label_ns_map[resource_on_node]["label"],
+                    resource_count=1,
+                )
+            nodes_multicluster[index].start_nodes([resource_node])
+            wait_for_nodes_status([resource_node.name])
+            logger.info(
+                "Wait for all the pods in openshift-storage to be in running state"
+            )
+            assert wait_for_pods_to_be_running(
```
Comment (keesturam) [inline tests/disaster-recovery/regional-dr/test_node_failure.py:117]: Do we run IO and validate consistency with replication intervals? if not, pls consider adding them in a subsequent PR

File: tests/disaster-recovery/regional-dr/test_managed_cluster_node_failure.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+                node_names=[resource_node.name], status=constants.NODE_NOT_READY
+            )
+
+            # Wait for pod to be rescheduled. For submariner-gateway, use sleep as it doesn't get rescheduled
+            if resource_on_node == "submariner-gateway":
+                logger.info("Waiting for 300 seconds before starting the node")
+                time.sleep(300)
+            else:
+                # Wait for pod to reach terminating state or to be deleted
+                assert wait_for_pods_to_be_in_statuses(
```
Comment (am-agrawa) [inline tests/disaster-recovery/regional-dr/test_managed_cluster_node_failure.py:106]: The comment says we are checking multiple statuses but I could see only one condition for Terminating.

**Review by am-agrawa — APPROVED** [review]
Other than missing log.info at a few places, everything else looks good.

File: tests/disaster-recovery/regional-dr/test_managed_cluster_node_failure.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        },
+    }
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        def finalizer():
+            for cluster in get_non_acm_cluster_config():
+                config.switch_ctx(cluster.MULTICLUSTER["multicluster_index"])
+                logger.info("Checking for Ceph Health OK")
+                ceph_health_check(tries=40, delay=60)
```
Comment (ebenahar) [inline tests/disaster-recovery/regional-dr/test_managed_cluster_node_failure.py:64]: we should also bring up any node that was left down if the test fails

File: tests/disaster-recovery/regional-dr/test_managed_cluster_node_failure.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+@tier4
+@tier4b
+@pytest.mark.parametrize(
+    argnames=["resource_on_node"],
+    argvalues=[
+        pytest.param(
+            "ramen-dr-cluster-operator", marks=pytest.mark.polarion_id("OCS-4438")
+        ),
+        pytest.param("rbd-mirror", marks=pytest.mark.polarion_id("OCS-4437")),
+        pytest.param("submariner-gateway", marks=pytest.mark.polarion_id("OCS-4439")),
```
Comment (am-agrawa) [inline tests/disaster-recovery/regional-dr/test_managed_cluster_node_failure.py:33]: I think it's good to add scenarios for other submariner pods too, as we have been seeing a lot of connectivity issues due to submariner pods which may not be specific to gateway only.

**Pull request merged**